addons/arsenal/functions/fnc_updateUniqueItemsList.sqf

```diff
@@ -13,106 +14,166 @@
-            ((GVAR(virtualItems) select 22) select 2) pushBackUnique _x;
-        };
-        // acc
-        case (
-                isClass (_configCfgWeapons >> _x) &&
-                {!(_x in ((GVAR(virtualItems) select IDX_VIRT_ATTACHEMENTS) select 3))} &&
-                {_x in ((_itemsCache select 1) select 3)}
-            ): {
-            ((GVAR(virtualItems) select 22) select 3) pushBackUnique _x;
+            switch (true) do {
+                // "Misc. items" magazines (e.g. spare barrels, intel, photos)
+                case (
+                    !(_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS)) &&
+                    {_x in (uiNamespace getVariable QGVAR(magazineMiscItems)) ||
+                    {getNumber (_config >> "ACE_isUnique") == 1}}
+                ): {
+                    (GVAR(virtualItems) get IDX_VIRT_UNIQUE_MISC_ITEMS) set [_x, nil];
+                };
+                // Primary, Handgun, Secondary weapon magazines
+                case (
+                    !(_x in (GVAR(virtualItems) get IDX_VIRT_ITEMS_ALL)) &&
+                    {_x in (_configItems get IDX_VIRT_ITEMS_ALL) ||
+                    {getNumber (_config >> QGVAR(hide)) == -1} ||
+                    {getNumber (_config >> "type") in [TYPE_MAGAZINE_PRIMARY_AND_THROW, TYPE_MAGAZINE_SECONDARY_AND_PUT, 1536, TYPE_MAGAZINE_HANDGUN_AND_GL, TYPE_MAGAZINE_MISSILE]}}
+                ): {
+                    (GVAR(virtualItems) get IDX_VIRT_UNIQUE_VIRT_ITEMS_ALL) set [_x, nil];
+                };
+                // Grenades
+                case (
+                    !(_x in (GVAR(virtualItems) get IDX_VIRT_GRENADES)) &&
+                    {_x in (uiNamespace getVariable QGVAR(grenadeCache))}
+                ): {
+                    (GVAR(virtualItems) get IDX_VIRT_UNIQUE_GRENADES) set [_x, nil];
+                };
+                // Explosives
+                case (
+                    !(_x in (GVAR(virtualItems) get IDX_VIRT_EXPLOSIVES)) &&
+                    {_x in (uiNamespace getVariable QGVAR(putCache))}
+                ): {
+                    (GVAR(virtualItems) get IDX_VIRT_UNIQUE_EXPLOSIVES) set [_x, nil];
+                };
+                // Unknown
+                default {
+                    // Don't add items that are part of the arsenal
+                    if !(_x in GVAR(virtualItemsFlatAll)) then {
+                        (GVAR(virtualItems) get IDX_VIRT_UNIQUE_UNKNOWN_ITEMS) set [_x, nil];
+                    };
+                };
+            };
         };
+        // Weapons
+        case (_isWeapon): {
+            _config = _cfgWeapons >> _x;
+            _simulationType = getText (_config >> "simulation");
+            _configItemInfo = _config >> "ItemInfo";
+            _hasItemInfo = isClass (_configItemInfo);
+            _itemInfoType = if (_hasItemInfo) then {getNumber (_configItemInfo >> "type")} else {0};
+            _isMiscItem = _className isKindOf ["CBA_MiscItem", _cfgWeapons];
```

Comment:
`_className` is undefined, probably just `_x` but not sure
```suggestion
            _isMiscItem = _x isKindOf ["CBA_MiscItem", _cfgWeapons];
```

Reply:
`_x` is classname. That works.